src/webserver/SslWsSession.h

```diff
@@ -65,82 +49,38 @@ class SslWsSession : public WsSession<SslWsSession>
     {
         return ws_;
     }
-
-    std::optional<std::string>
-    ip()
-    {
-        return ip_;
-    }
 };
 
-class SslWsUpgrader : public std::enable_shared_from_this<SslWsUpgrader>
+/**
+ * @brief The SSL WebSocket upgrader class, upgrade from http session to websocket session.
+ */
+template <ServerHandler Handler>
+class SslWsUpgrader : public std::enable_shared_from_this<SslWsUpgrader<Handler>>
 {
-    boost::asio::io_context& ioc_;
     boost::beast::ssl_stream<boost::beast::tcp_stream> https_;
     boost::optional<http::request_parser<http::string_body>> parser_;
     boost::beast::flat_buffer buffer_;
-    std::optional<std::string> ip_;
-    std::shared_ptr<BackendInterface const> backend_;
-    std::shared_ptr<RPC::RPCEngine> rpcEngine_;
-    std::shared_ptr<SubscriptionManager> subscriptions_;
-    std::shared_ptr<LoadBalancer> balancer_;
-    std::shared_ptr<ETLService const> etl_;
-    util::TagDecoratorFactory const& tagFactory_;
-    clio::DOSGuard& dosGuard_;
+    std::string ip_;
+    std::reference_wrapper<util::TagDecoratorFactory const> tagFactory_;
+    std::reference_wrapper<clio::DOSGuard> dosGuard_;
+    std::shared_ptr<Handler> const handler_;
     http::request<http::string_body> req_;
 
 public:
     SslWsUpgrader(
-        boost::asio::io_context& ioc,
-        std::optional<std::string> ip,
-        boost::asio::ip::tcp::socket&& socket,
-        ssl::context& ctx,
-        std::shared_ptr<BackendInterface const> backend,
-        std::shared_ptr<RPC::RPCEngine> rpcEngine,
-        std::shared_ptr<SubscriptionManager> subscriptions,
-        std::shared_ptr<LoadBalancer> balancer,
-        std::shared_ptr<ETLService const> etl,
-        util::TagDecoratorFactory const& tagFactory,
-        clio::DOSGuard& dosGuard,
-        boost::beast::flat_buffer&& b)
-        : ioc_(ioc)
-        , https_(std::move(socket), ctx)
-        , buffer_(std::move(b))
-        , ip_(ip)
-        , backend_(backend)
-        , rpcEngine_(rpcEngine)
-        , subscriptions_(subscriptions)
-        , balancer_(balancer)
-        , etl_(etl)
-        , tagFactory_(tagFactory)
-        , dosGuard_(dosGuard)
-    {
-    }
-
-    SslWsUpgrader(
-        boost::asio::io_context& ioc,
         boost::beast::ssl_stream<boost::beast::tcp_stream> stream,
-        std::optional<std::string> ip,
-        std::shared_ptr<BackendInterface const> backend,
-        std::shared_ptr<RPC::RPCEngine> rpcEngine,
-        std::shared_ptr<SubscriptionManager> subscriptions,
-        std::shared_ptr<LoadBalancer> balancer,
-        std::shared_ptr<ETLService const> etl,
-        util::TagDecoratorFactory const& tagFactory,
-        clio::DOSGuard& dosGuard,
+        std::string ip,
+        std::reference_wrapper<util::TagDecoratorFactory const> tagFactory,
+        std::reference_wrapper<clio::DOSGuard> dosGuard,
+        std::shared_ptr<Handler> const& handler,
         boost::beast::flat_buffer&& b,
```

Comment:
`b` is a bad name (i know it is from the old code). Can we update it to `buf` or `buffer` please? ⭐